api/BaseAttribute.py

```diff
@@ -0,0 +1,70 @@
+
+"""
+    Copyright (c) 2015-2016 Cisco Systems, Inc. and others.  All rights reserved.
+    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License v1.0 which accompanies this distribution,
+    and is available at http:#www.eclipse.org/legal/epl-v10.html
+"""
+
+from Base import *
+from FieldProcessors import *
+
+class BaseAttribute(Base):
+    """
+    Format class for base_attribute parsed messages (openbmp.parsed.base_attribute)
+
+    Schema Version: 1.2
+    """
+
+    def __init__(self, data):
+        """
+        Handle the message by parsing it and storing the data in memory.
+
+        :param data: Data to parse.
+        """
+
+        super(BaseAttribute, self).__init__()
+
+        self.headerNames = ["action", "seq", "hash", "router_hash", "router_ip", "peer_hash", "peer_ip",
```

Comment:
Let's change all of these to use an enum/class for fields.  Moody is working on this change for Java as well where he has a fields schema class that defines all fields by code accessor which can then render the field name and value.   For example:

MsgBusFields.ACTION.getName() to get the header name and MsgBusFields.ACTION.getDefaultValue() to get the default value if not present in the TSV. We need this in code to avoid errors with people typo'ing the dictionary keys.  We also need this so that default values can be populated when supporting older schemas where the fields are not present.